lib/endpoint.ex

```diff
@@ -42,7 +42,7 @@ defmodule SparkPost.Endpoint do
       else
         Map.merge(headers, %{"Content-Type": "application/json"})
       end
-      |> Map.merge(base_request_headers)
+      |> Map.merge(base_request_headers())
```

Comment:
I see this pattern of adding parentheses on /0 func calls throughout the PR. Can you educate me a little on the need for this?